app/classes/class_knights.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+
+class Knights:
+
+    def __init__(self, knight: dict) -> None:
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.protection = 0
+
+        # preparations for battle
+        self.protection_calc(knight["armour"])
+        self.weapon_power(knight["weapon"])
+        self.drink_potion(knight["potion"])
```

Comment:
It's a good practice to separate different parts of your code into separate modules to improve code organization, maintainability, and readability. Separating the battle-related logic into another module, such as a `battle` module, can make your codebase cleaner and more modular

Reply:
I've added new "battle" package with "preparation" and "fight" modules.